src/shared/services/shortcut-action.service.ts

```diff
@@ -0,0 +1,276 @@
+    }
+   }
+
+  upAndDownAction(path: Array<any>, direction: number) {
+    let values = this.jsonStoreService.getIn(this.pathUtilService.getNearestOrRootArrayParentInPath(path, false));
+    if (List.isList(values)) {
+      let elemIndexInPath = this.pathUtilService.getElementIndexInPath(path);
+      path[path.length - 2] = ((elemIndexInPath + direction) < values.size &&
+                              (elemIndexInPath + direction) >= 0) ? elemIndexInPath + direction :
+                              values.size - Math.abs((elemIndexInPath + direction));
+      this.domUtilService.focusAndSelectFirstInputChildById(path.join('.'));
+    }
+  }
+
+  rightAndLeftAction(path: Array<any>, direction: number) {
+    let tabIndex = this.tabIndexService.getElemTabIndex(path.join('.'));
+    if (tabIndex >= 0) {
+      this.domUtilService.focusRightOrLeftCell(tabIndex, direction);
+    }
+  }
+
+  copyCurrentRowBelowAction(path: Array<any>) {
+    let _path = this.pathUtilService.getNearestOrRootArrayParentInPath(path, false);
+    this.copyCurrentRowBelow(path, _path, this.jsonSchemaService.getSchemaFromPath(_path));
+  }
+
+  copyCurrentSchemaItemBelowAction(path: Array<any>) {
+    let _path = this.pathUtilService.getNearestOrRootArrayParentInPath(path, true);
+    this.copyCurrentSchemaItemBelow(path, _path, this.jsonSchemaService.getSchemaFromPath(_path));
+  }
+
+  private copyCurrentRowBelow(originalPath: Array<any>, arrayParentPath: Array<any>, schema: Object) {
+    let elemIndex = this.pathUtilService.getElementIndexInReversePath(originalPath);
+    let valuesList = this.jsonStoreService.getIn(arrayParentPath) || List.of([]);
+    let newValue = valuesList.get(elemIndex);
+    newValue = newValue.set(originalPath[originalPath.length - 1], '');
+    this.jsonStoreService.setIn(arrayParentPath, valuesList.insert(elemIndex + 1, newValue));
+    arrayParentPath.push(elemIndex + 1);
+    setTimeout(() => {
+      this.updateTabIndexesAndFocusElementInPath(arrayParentPath.join('.').concat('.' + originalPath[originalPath.length - 1]));
+    });
+  }
+
+  private copyCurrentSchemaItemBelow(originalPath: Array<any>, arrayParentPath: Array<any>, schema: Object) {
+    let elemIndex = this.pathUtilService.getElementIndexInPath(originalPath);
+    let valuesList = this.jsonStoreService.getIn(arrayParentPath) || List.of([]);
+    let newValue = valuesList.get(elemIndex);
+    this.jsonStoreService.setIn(arrayParentPath, valuesList.insert(elemIndex + 1, newValue));
+    arrayParentPath.push(elemIndex + 1);
+    setTimeout(() => {
+      this.updateTabIndexesAndFocusElementInPath(arrayParentPath.join('.'));
+    });
+  }
+
+  private updateTabIndexesAndFocusElementInPath(path: string) {
+    this.tabIndexService.insertElemTabIndex(path);
+    this.domUtilService.flashElementById(path);
+    this.domUtilService.focusAndSelectFirstInputChildById(path);
+  }
+
+  generateShortcutAction(action: string) {
```

Comment:
@harunurhan this generator function will return every callback function based on given action. Is it ok?